sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -194,12 +193,18 @@ public SeekableByteChannel newByteChannel(Path path, Set<? extends OpenOption> s
     }
 
     /**
+     * Returns an {@link AzureDirectoryStream} for iterating over the contents of a directory.
+     *
      * {@inheritDoc}
      */
     @Override
     public DirectoryStream<Path> newDirectoryStream(Path path, DirectoryStream.Filter<? super Path> filter)
         throws IOException {
-        return null;
+        if (!(path instanceof AzurePath)) {
+            throw LoggingUtility.logError(logger, new IllegalArgumentException("This provider cannot operate on "
```

Comment:
limitation on path type should be mentioned in javadoc I think and what is expected if somebody passes something other than AzurePath.

Reply:
So I've been doing this defensively just to be safe, but I don't think under normal circumstances paths usually get passed to providers other than their own. People usually work through the Files type, which gets the appropriate provider for the given path and then passes it to the method on the provider internally. To screw this up, someone would have to query for the provider themselves and make an explicit all. 

The docs also aren't super clear to me on what the behavior should be if a path is used with a provider that is not its own. I suppose instead of asserting the right type, I could always just get the string representation of the path and construct an AzurePath from that. Thoughts?

Reply:
I'm questioning that validation. I'm just saying that we should have `@throws` section in javadoc for this method explaining that behavior - which is not obvious by just reading method signature (it promises to take any `Path`).

Whether to keep validation or try to convert other Paths. I think going with validation is good start. It's easier to loosen it later than make it strict, so let's be strict and wait for feedback.